File: src/2021/day01/day01.kt
import java.io.File

fun readFile(file: String): List<Int> {
    return File(file)
            .readLines()
            .map(String::toInt)
}

fun part1(file: String): Int {
    val numbers = readFile(file)
    var curr = numbers[0]
    var counter = 0
    for (num in numbers) {
        if (num > curr) {
            counter += 1
        }
        curr = num
    }
    return counter
}

fun part2(file: String): Int {
    val numbers = readFile(file)
    var curr = numbers[0] + numbers[1] + numbers[2]
    var counter = 0
    for (i in numbers.indices) {
        if (i < 3) {
            continue
        }
        val next = curr - numbers[i-3] + numbers[i]
        if (next > curr) {
            counter += 1
        }
        curr = next
    }
    return counter
}

fun main() {
    check(part1("test") == 7)
    println(part1("input"))
    check(part2("test") == 5)
    println(part2("input"))
}

File: src/2022/day01/day01.py
def read_file(name):
  with open(name) as file:
    lines = [line.rstrip() for line in file]
  return lines


def part1(name):
  lines = read_file(name)
  elves = []
  curr = 0
  for line in lines:
    if not line:
      elves.append(curr)
      curr = 0
    else:
      curr += int(line)
  return max(elves)


def part2(name):
  lines = read_file(name)
  elves = []
  curr = 0
  for line in lines:
    if not line:
      elves.append(curr)
      curr = 0
    else:
      curr += int(line)
  elves = sorted(elves, reverse=True)
  return elves[0] + elves[1] + elves[2]


print('Part 1')
print(part1('test'))
print(part1('input'))

print('Part 2')
print(part2('test'))
print(part2('input'))
